pkg/agent/docker_secrets_postrenderer.go

```diff
@@ -0,0 +1,57 @@
+package agent
+
+import (
+	"bytes"
+	"io"
+
+	"gopkg.in/yaml.v3"
```

Comment:
It was just to use the latest. The only difference that's relevant is that v3 has a default tab intent of 4 spaces, as well as a `SetIndent()` function. I'll switch to v2 since I'd not realised v3 isn't yet stable/released.